src/Libraries/Microsoft.Extensions.Diagnostics.ResourceMonitoring/Windows/Interop/MemoryInfo.cs

```diff
@@ -10,20 +10,16 @@ namespace Microsoft.Extensions.Diagnostics.ResourceMonitoring.Windows.Interop;
 /// Native memory interop methods.
 /// </summary>
 [ExcludeFromCodeCoverage]
-internal sealed class MemoryInfo : IMemoryInfo
+internal sealed partial class MemoryInfo : IMemoryInfo
 {
-    internal MemoryInfo()
-    {
-    }
-
     /// <summary>
     /// Get the memory status of the host.
     /// </summary>
     /// <returns>Memory status information.</returns>
-    public unsafe MEMORYSTATUSEX GetMemoryStatus()
+    public MEMORYSTATUSEX GetMemoryStatus()
     {
         MEMORYSTATUSEX info = default;
-        info.Length = (uint)sizeof(MEMORYSTATUSEX);
+        info.Length = (uint)Marshal.SizeOf<MEMORYSTATUSEX>();
```

Comment:
This change has negative perf characteristics.
If not already defined [`MEMORYSTATUSEX`](https://learn.microsoft.com/windows/win32/api/sysinfoapi/ns-sysinfoapi-memorystatusex) should be defined as a blittable struct to avoid all associated marshaling.
https://github.com/dotnet/runtime/blob/31733b9a35185785427bac69ef80a4eb56b727c2/src/libraries/Common/src/Interop/Windows/Kernel32/Interop.MEMORYSTATUSEX.cs

Reply:
MEMORYSTATUSEX is already defined in the MemoryStatusEx.cs file. Is this what you mean?

Reply:
Reverted back :)